nullaway/src/main/java/com/uber/nullaway/ErrorBuilder.java

```diff
@@ -198,10 +200,13 @@ private Description.Builder addSuggestedSuppression(
       case PASS_NULLABLE:
       case ASSIGN_FIELD_NULLABLE:
       case SWITCH_EXPRESSION_NULLABLE:
-        if (config.getCastToNonNullMethod() != null) {
+        if (config.getCastToNonNullMethod() != null && canBeCastToNonNull(suggestTree)) {
           builder = addCastToNonNullFix(suggestTree, builder);
         } else {
-          builder = addSuppressWarningsFix(suggestTree, builder, suppressionName);
+          Tree suppressibleNode = suppressibleNode(state.getPath());
```

Comment:
This feels like it should work, but what was the reasoning for not using `suggestTree` here and instead querying `state.getPath()` again?

Reply:
If `config.getCastToNonNullMethod() != null` then `suggestTree` changes and becomes the `@Nullable` argument to be casted (see [here](https://github.com/uber/NullAway/blob/56d5ff142a3cacc5ee4b9be7137d099befe0407d/nullaway/src/main/java/com/uber/nullaway/ErrorBuilder.java#L249-L252)).  With this PR, sometimes even if the `castToNonNull` method is present, we may not use it and end up in the `else` case above.  For those cases, the `suggestTree` is no longer suppressible and we need to call `suppressibleNode()`.  For cases where `config.getCastToNonNullMethod() == null` this call to `suppressibleNode()` is redundant.  Do you think it's sufficient to add a comment explaining this?  Or do we want to try to remove the redundant call (which will make the logic more complicated)?

Reply:
Makes sense :) Up to you if it even needs a comment, I don't think the question occurs the reader unless they see the before and after code both.

Reply:
Added a comment